code/items/ammo/TTTAmmo.cs

```diff
@@ -0,0 +1,92 @@
+using System;
+using Sandbox;
+using TTTReborn.Player;
+
+namespace TTTReborn.Items
+{
+    [Library("ttt_ammo"), Hammer.Skip]
+    public abstract partial class TTTAmmo : Prop
+    {
+        /// <summary>
+        /// String definition of ammo type, should match TTTWeapon.AmmoType
+        /// </summary>
+        public virtual string AmmoType { get; set; }
+        /// <summary>
+        /// Amount of Ammo within Entity.
+        /// </summary>
+        public virtual int AmmoAmount { get; set; }
+
+        /// <summary>
+        /// Maximum amount of ammo player can carry in their inventory.
+        /// </summary>
+        public virtual int AmmoMax { get; set; }
+
+        [Net]
+        private int CurrentAmmo { get; set; }
+        private int AmmoEntMax { get; set; }
+
+        public virtual string ModelPath => "models/ammo_buckshot.vmdl";
+
+        public override void Spawn()
+        {
+            base.Spawn();
+
+            SetModel(ModelPath);
+            SetupPhysicsFromModel(PhysicsMotionType.Dynamic);
+            RemoveCollisionLayer(CollisionLayer.Player);
+
+            AmmoEntMax = AmmoAmount;
+            CurrentAmmo = AmmoAmount;
+        }
+
+        public override void Touch(Entity other)
+        {
+            base.Touch(other);
+            if (IsServer)
+            {
+                if(other is TTTPlayer player)
+                {
+                    var ammoType = AmmoType.ToLower();
```

Comment:
Is this `ToLower()` call necessary?

Reply:
All inputs into the ammo inventory system are `ToLower`ed. Will stop issues of new ammo types being created if someone writes `'SMG'` rather than `'smg'`. With how it is currently, capitalization coming from the Weapon and Ammo definitions are all lowered and kept consistent in that manner.

I can't imagine there would be a need for an ammo type with different capitalization that cannot just be called something else instead. Less confusing if case sensitivity is not a factor.